app/routes/verify.js

```diff
@@ -0,0 +1,27 @@
+import Route from '@ember/routing/route';
+import UnauthenticatedRouteMixin from 'ember-simple-auth/mixins/unauthenticated-route-mixin';
+
+export default Route.extend(UnauthenticatedRouteMixin, {
+  titleToken() {
+    return this.get('l10n').t('Verify');
+  },
+
+  beforeModel(transition) {
+    let payload = {
+      data: {
+        token: transition.queryParams.token
+      }
+    };
+
+    return this.get('loader')
+      .post('auth/verify-email', payload)
+      .then(() => {
+        this.controllerFor('verify').message = 'Your email has been verified successfully! Please login to continue.';
+        this.controllerFor('verify').success = true;
+      })
+      .catch(() => {
+        this.controllerFor('verify').message = 'An unexpected error occured.';
```

Comment:
Don't say that an unexpected error occured. Show them the real error message response sent from the server.